packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -52,25 +56,38 @@ export class TelemetryBuilder {
   }
 }
 
+// export only for unit test
+export class TelemetryServiceProvider {
+  public static instances = new Map<string, TelemetryService>(); // public only for unit test
+  public static getInstance(extensionName?: string): TelemetryService {
+    // default if not present
+    const name = extensionName || SFDX_CORE_EXTENSION_NAME;
+    if (!TelemetryServiceProvider.instances.has(name)) {
+      TelemetryServiceProvider.instances.set(name, new TelemetryService());
+    }
+    return TelemetryServiceProvider.instances.get(name)!;
+  }
+}
+
 export class TelemetryService {
-  private static instance: TelemetryService;
   private extensionContext: ExtensionContext | undefined;
   private reporter: TelemetryReporter | undefined;
   private aiKey = DEFAULT_AIKEY;
   private version: string = '';
+  /**
+   * Retrieve Telemetry Service according to the extension name.
+   * If no extension name provided, return the instance for core extension by default
+   * @param extensionName extension name
+   */
+  public static getInstance(extensionName?: string) {
+    return TelemetryServiceProvider.getInstance(extensionName);
+  }
   /**
    * Cached promise to check if CLI telemetry config is enabled
    */
   private cliAllowsTelemetryPromise?: Promise<boolean> = undefined;
   public extensionName: string = 'unknown';
 
-  public static getInstance() {
-    if (!TelemetryService.instance) {
-      TelemetryService.instance = new TelemetryService();
-    }
-    return TelemetryService.instance;
-  }
-
   /**
    * Initialize Telemetry Service during extension activation.
    * @param extensionContext extension context
```

Comment:
Also, on line 107 `this.extensionName` is set to `name`

Since this.extensionName is set in the getInstance method, and used as the key in the map in the Provider, then the extensionName property should probably not be renamed.  What do you think about a future work item, after both of these prs are merged and released, to throw an error if the Telem Service is initialized with a different context?  Something like:
<img width="739" alt="2023-11-27_22-10-08" src="https://github.com/forcedotcom/salesforcedx-vscode/assets/46458081/96bd94de-7fec-425a-8da9-c14f896e3e72">